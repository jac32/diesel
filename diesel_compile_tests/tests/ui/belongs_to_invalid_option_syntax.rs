#[macro_use]
extern crate diesel;

table! {
    foo {
        id -> Integer,
    }
}

#[derive(Identifiable)]
#[table_name = "foo"]
struct Bar {
    id: i32,
}

#[derive(Identifiable)]
#[table_name = "foo"]
struct Baz {
    id: i32,
}

#[derive(Associations)]
#[belongs_to]
#[belongs_to = "Bar"]
#[belongs_to()]
#[belongs_to(foreign_key = "bar_id")]
#[belongs_to(Bar, foreign_key)]
#[belongs_to(Bar, foreign_key(bar_id))]
#[belongs_to(Baz, foreign_key = "bar_id", random_option)]
struct Foo {
    bar_id: i32,
}

fn main() {}
