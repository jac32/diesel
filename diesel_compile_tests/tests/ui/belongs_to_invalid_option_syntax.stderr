error: `belongs_to` must be in the form `belongs_to(...)`
 --> <macro expansion>:1:1
  |
1 | #[belongs_to]
  | ^

error: `belongs_to` must be in the form `belongs_to(...)`
 --> <macro expansion>:1:1
  |
1 | #[belongs_to = "Bar"]
  | ^

error: Expected a struct name
 --> <macro expansion>:1:1
  |
1 | #[belongs_to()]
  | ^
  |
  = help: e.g. `#[belongs_to(User)]`

error: Expected a struct name
 --> <macro expansion>:1:1
  |
1 | #[belongs_to(foreign_key = "bar_id")]
  | ^
  |
  = help: e.g. `#[belongs_to(User)]`

error: `foreign_key` must be in the form `foreign_key = "value"`
 --> <macro expansion>:1:1
  |
1 | #[belongs_to(Bar, foreign_key)]
  | ^

warning: The form `foreign_key(value)` is deprecated. Use `foreign_key = "value"` instead
 --> <macro expansion>:1:1
  |
1 | #[belongs_to(Bar, foreign_key(bar_id))]
  | ^

warning: Unrecognized option random_option
 --> <macro expansion>:1:1
  |
1 | #[belongs_to(Baz, foreign_key = "bar_id", random_option)]
  | ^

error: aborting due to 5 previous errors

